library/built-in/_prefabs/prefab_cnet.ts

```diff
@@ -98,108 +98,101 @@ export type Cnet_return = {
-                ckpt_return = (await ip_adapter_result).ip_adapted_model
+                const ip_adapter_result = run_cnet_IPAdapter(cnetImage.cnets.IPAdapter, ctx, image)
+                args.ckptPos = ip_adapter_result.ip_adapted_model
             } else {
-                let strength: number | undefined
                 // CANNY ===========================================================
-                if (cnetImage.cnets.Canny) {
-                    const cnet_return_canny = await run_cnet_canny(cnetImage.cnets.Canny, cnet_args, image)
-                    strength = cnetImage.cnets.Canny.strength
-                    image = cnet_return_canny.image
-                    cnet_name = cnet_return_canny.cnet_name
+                const { Canny, Depth, Normal, Lineart, OpenPose, Scribble, SoftEdge, Tile } = cnetImage.cnets
+                if (Canny) {
+                    const y = run_cnet_canny(Canny, ctx, image)
+                    _apply_cnet(args, Canny.strength, y.image, y.cnet_name)
                 }
                 // POSE ===========================================================
-                else if (cnetImage.cnets.OpenPose) {
-                    const cnet_return_openPose = await run_cnet_openPose(cnetImage.cnets.OpenPose, cnet_args, image)
-                    strength = cnetImage.cnets.OpenPose.strength
-                    image = cnet_return_openPose.image
-                    cnet_name = cnet_return_openPose.cnet_name
+                if (OpenPose) {
+                    const y = run_cnet_openPose(OpenPose, ctx, image)
+                    _apply_cnet(args, OpenPose.strength, y.image, y.cnet_name)
                 }
                 // DEPTH ===========================================================
-                else if (cnetImage.cnets.Depth) {
-                    const cnet_return_depth = await run_cnet_Depth(cnetImage.cnets.Depth, cnet_args, image)
-                    strength = cnetImage.cnets.Depth.strength
-                    image = cnet_return_depth.image
-                    cnet_name = cnet_return_depth.cnet_name
+                if (Depth) {
+                    const y = run_cnet_Depth(Depth, ctx, image)
+                    _apply_cnet(args, Depth.strength, y.image, y.cnet_name)
                 }
                 // Normal ===========================================================
-                else if (cnetImage.cnets.Normal) {
-                    const cnet_return_normal = await run_cnet_Normal(cnetImage.cnets.Normal, cnet_args, image)
-                    strength = cnetImage.cnets.Normal.strength
-                    image = cnet_return_normal.image
-                    cnet_name = cnet_return_normal.cnet_name
+                if (Normal) {
+                    const y = run_cnet_Normal(Normal, ctx, image)
+                    _apply_cnet(args, Normal.strength, y.image, y.cnet_name)
```

Comment:
those 2 lines copy pasted are not perfect, but I'd say acceptable for now, the same way previous few lines where copy pasted. There are either probably better names to find, or maybe the _apply_cnet should be called directly within run_cnet_Normal ?